### PR TITLE
python3Packages.graphtage: init at 0.2.5

### PR DESCRIPTION
**Pull request by veehaitch**
###### Motivation for this change

A utility to diff tree-like files such as JSON and XML

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114546` at 4589100c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/3bf72640f7950268fd0e8f4112813a8d">python37Packages.graphtage</a></li></li>
    <li><a href="https://gist.github.com/3c283ea6ff8e264f08e83d9c79b1432a">python37Packages.json5</a></li></li>
    <li><a href="https://gist.github.com/dc8ef253b988003f8c889d56dd2cd11d">python38Packages.graphtage</a></li></li>
    <li><a href="https://gist.github.com/91bedac3a7505be859fa9e22e1b8da88">python38Packages.json5</a></li></li>
    <li><a href="https://gist.github.com/22e5fc11ec1d67ce3b0dcbed12ff1c50">python39Packages.graphtage</a></li></li>
    <li><a href="https://gist.github.com/600d159eab0cfc9313e458e198722f00">python39Packages.json5</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/graphtage/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  };
+
+  propagatedBuildInputs = [
+    colorama
+    intervaltree
+    json5
+    pyyaml
+    scipy
+    tqdm
+    typing-extensions
```
[the remark this answers is not in the document]
Comment (veehaitch) [reply pkgs/development/python-modules/graphtage/default.nix:35]: I believe, it's required, yes. If I strip it from `setup.py`, testing fails using 3.9

File: pkgs/development/python-modules/graphtage/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+, pyyaml
+, scipy
+, tqdm
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "graphtage";
+  version = "0.2.5";
+  disabled = isPy27;
```
[the remark this answers is not in the document]
Comment (veehaitch) [reply pkgs/development/python-modules/graphtage/default.nix]: Thanks, done!

File: pkgs/development/python-modules/graphtage/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    colorama
+    intervaltree
+    json5
+    pyyaml
+    scipy
+    tqdm
+    typing-extensions
+  ];
+
+  checkInputs = [ pytestCheckHook ];
```
[the remark this answers is not in the document]
Comment (veehaitch) [reply pkgs/development/python-modules/graphtage/default.nix:38]: Added.

File: pkgs/development/python-modules/graphtage/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/graphtage/default.nix:35]: most likely an upstream issue.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 114546 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>6 packages built:</summary>  <ul>    <li>python37Packages.graphtage</li>    <li>python37Packages.json5</li>    <li>python38Packages.graphtage</li>    <li>python38Packages.json5</li>    <li>python39Packages.graphtage</li>    <li>python39Packages.json5</li>  </ul></details>

File: pkgs/development/python-modules/graphtage/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, pytestCheckHook
+, pythonOlder
+  # Python dependencies
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/graphtage/default.nix:6]: This comment can be removed.

File: pkgs/development/python-modules/json5/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "dpranke";
+    repo = "pyjson5";
+    rev = "v${version}";
+    sha256 = "sha256-VkJnZG1BuC49/jJuwObbqAF48CtbWU9rDEYW4Dg0w4U=";
+  };
+
+  checkInputs = [
+    hypothesis
+    pytestCheckHook
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/json5/default.nix:22]: Missing pythonImportsCheck.

**Pull request merged**